routing/triert/table_test.go

```diff
@@ -0,0 +1,375 @@
+		require.NoError(t, err)
+		require.True(t, success)
+
+		require.Equal(t, 2, rt.Size())
+		require.Equal(t, 2, rt.CplSize(2))
+
+		require.Equal(t, 2, rt.CplSize(1))
+		require.Equal(t, 0, rt.CplSize(3))
+	})
+
+	t.Run("cpl 3", func(t *testing.T) {
+		ctx := context.Background()
+		rt := New(key0)
+
+		success, err := rt.AddPeer(ctx, NewNode("cpl3a", keyWithPrefix("0001", 32)))
+		require.NoError(t, err)
+		require.True(t, success)
+		success, err = rt.AddPeer(ctx, NewNode("cpl3b", keyWithPrefix("0001", 32)))
+		require.NoError(t, err)
+		require.True(t, success)
+		success, err = rt.AddPeer(ctx, NewNode("cpl3c", keyWithPrefix("0001", 32)))
+		require.NoError(t, err)
+		require.True(t, success)
+		success, err = rt.AddPeer(ctx, NewNode("cpl3d", keyWithPrefix("0001", 32)))
+		require.NoError(t, err)
+		require.True(t, success)
+
+		require.Equal(t, 4, rt.Size())
+		require.Equal(t, 4, rt.CplSize(3))
+
+		require.Equal(t, 4, rt.CplSize(1))
+		require.Equal(t, 4, rt.CplSize(2))
+	})
+
+	t.Run("cpl mixed", func(t *testing.T) {
+		ctx := context.Background()
+		rt := New(key0)
+
+		success, err := rt.AddPeer(ctx, NewNode("cpl1a", keyWithPrefix("01", 32)))
+		require.NoError(t, err)
+		require.True(t, success)
+		success, err = rt.AddPeer(ctx, NewNode("cpl1b", keyWithPrefix("01", 32)))
+		require.NoError(t, err)
+		require.True(t, success)
+
+		success, err = rt.AddPeer(ctx, NewNode("cpl2a", keyWithPrefix("001", 32)))
+		require.NoError(t, err)
+		require.True(t, success)
+		success, err = rt.AddPeer(ctx, NewNode("cpl2b", keyWithPrefix("001", 32)))
+		require.NoError(t, err)
+		require.True(t, success)
+
+		success, err = rt.AddPeer(ctx, NewNode("cpl3a", keyWithPrefix("0001", 32)))
+		require.NoError(t, err)
+		require.True(t, success)
+		success, err = rt.AddPeer(ctx, NewNode("cpl3b", keyWithPrefix("0001", 32)))
+		require.NoError(t, err)
+		require.True(t, success)
+		success, err = rt.AddPeer(ctx, NewNode("cpl3c", keyWithPrefix("0001", 32)))
+		require.NoError(t, err)
+		require.True(t, success)
+		success, err = rt.AddPeer(ctx, NewNode("cpl3d", keyWithPrefix("0001", 32)))
+		require.NoError(t, err)
+		require.True(t, success)
+
+		require.Equal(t, 8, rt.Size())
+		require.Equal(t, 8, rt.CplSize(1))
+		require.Equal(t, 6, rt.CplSize(2))
+		require.Equal(t, 4, rt.CplSize(3))
+	})
+}
+
+func keyWithPrefix(s string, length int) key.KadKey {
+	bits := len(s)
+	if bits > 64 {
+		panic("keyWithPrefix: prefix too long")
+	}
+	n, err := strconv.ParseInt(s, 2, 64)
+	if err != nil {
+		panic("keyWithPrefix: " + err.Error())
+	}
+	prefix := uint64(n) << (64 - bits)
+
+	buf := make([]byte, length)
+	if _, err := rand.Read(buf); err != nil {
+		panic("keyWithPrefix: failed to read enough entropy for key")
+	}
+
+	lead := binary.BigEndian.Uint64(buf)
+	lead <<= bits
+	lead >>= bits
+	lead |= prefix
+	binary.BigEndian.PutUint64(buf, lead)
+	return key.KadKey(buf)
+}
+
+type node struct {
```

Comment:
`github.com/plprobelab/go-kademlia/network/address/kadid` could be used here. It was built for testing routing behaviors.

Reply:
I looked at it but it doesn't allow me to control the key and the string value independently. I wanted the strings to be readable for debugging tests.